auto-fallback.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const POSIX = `
+command_not_found_handler() {
+  # Do not run within a pipe
+  if test ! -t 1; then
+    echo "command not found: $1"
+    return 127
+  fi
+
+  echo "Trying with npx..."
+  npx $*
+  return $?
+}`
+
+const FISH = `
+function __fish_command_not_found_on_interactive --on-event fish_prompt
+  functions --erase __fish_command_not_found_handler
+  functions --erase __fish_command_not_found_setup
+
+  function __fish_command_not_found_handler --on-event fish_command_not_found
+    echo "Trying with npx..."
+    npx $argv
+  end
+
+  functions --erase __fish_command_not_found_on_interactive
+end`
+
+module.exports = function autoFallback (shell) {
+  const { BASH_VERSION, SHELL, ZSH_VERSION } = process.env
```

Comment:
We're restricted to `node@4` or latest, so was can't do destructuring assignment 😭